src/Web/Slack/Conversation.hs

```diff
@@ -181,69 +159,62 @@ data GroupConversation =
     -- , group_pending_shared :: []
     -- , group_pending_connected_team_ids :: []
 
-    , groupIsPendingExtShared :: Bool
-    , groupIsMember :: Bool
-    , groupIsPrivate :: Bool
-    , groupIsMpim :: Bool
-    , groupLastRead :: SlackTimestamp
-    , groupIsOpen :: Bool
-    , groupTopic :: Topic
-    , groupPurpose :: Purpose
-    , groupPriority :: Scientific
-    }
+    groupIsPendingExtShared :: Bool
+  , groupIsMember :: Bool
+  , groupIsPrivate :: Bool
+  , groupIsMpim :: Bool
+  , groupLastRead :: SlackTimestamp
+  , groupIsOpen :: Bool
+  , groupTopic :: Topic
+  , groupPurpose :: Purpose
+  , groupPriority :: Scientific
+  }
   deriving stock (Eq, Show, Generic)
 
 instance NFData GroupConversation
 
 $(deriveJSON (jsonOpts "group") ''GroupConversation)
 
-
 -- | Conversation object representing a (single-party) instance message,
 --   where only two people talk.
-data ImConversation =
-  ImConversation
-    { imId :: ConversationId
-    , imCreated :: Integer
-    , imIsArchived :: Bool
-    , imIsOrgShared :: Bool
-    , imUser :: UserId
-    , imIsUserDeleted :: Bool
-    , imPriority :: Scientific
-    }
+data ImConversation = ImConversation
+  { imId :: ConversationId
+  , imCreated :: Integer
+  , imIsArchived :: Bool
+  , imIsOrgShared :: Bool
+  , imUser :: UserId
+  , imIsUserDeleted :: Bool
+  , imPriority :: Scientific
+  }
   deriving stock (Eq, Show, Generic)
 
 instance NFData ImConversation
 
 $(deriveJSON (jsonOpts "im") ''ImConversation)
 
-
 -- | Ref. https://api.slack.com/types/conversation
---
---
-data Conversation =
-      Channel ChannelConversation
-    | Group GroupConversation
-    | Im ImConversation
+data Conversation
+  = Channel ChannelConversation
+  | Group GroupConversation
+  | Im ImConversation
   deriving stock (Eq, Show, Generic)
 
 instance NFData Conversation
 
-
 instance FromJSON Conversation where
   parseJSON = withObject "Conversation" $ \o ->
     parseWhen "is_channel" Channel o
       <|> parseWhen "is_group" Group o
       <|> parseWhen "is_im" Im o
       <|> prependFailure
-            "parsing a Conversation failed: neither channel, group, nor im, "
-            (typeMismatch "Conversation" (Object o))
-   where
-    parseWhen key con o = do
-      is <- (o .: key) <|> empty
-      if is
-        then con <$> parseJSON (Object o)
-        else empty
-
+        "parsing a Conversation failed: neither channel, group, nor im, "
+        (typeMismatch "Conversation" (Object o))
+    where
+      parseWhen key con o = do
+        is <- (o .: key)
```

Comment:
This is really the only real question I have on this PR. What effect does removing the `<|> empty` have?

So, I had to think about this. Either `o` has the key or it doesn't have the key.

If `o` has the key, then `o .: key` and `o .: key <|> empty` are the same, they both succeed and bind the value there to a `is :: Bool`. The subsequent code is the same in both versions, so no difference.

If `o` doesn't have the key, then `o .: key` short circuits the parser and possibly gives some context, whereas `o .: key <|> empty` short circuits the parser and provides no context. So the difference is that the new version `o .: key`, might have some additional context. The consuming code hasn't changed in this PR, so the new context, if any, isn't used in an essential way: it isn't branched on.

In summary, it looks like the new version, omitting `<|> empty`, provides at least as much information as the original version, but no new information is used downstream.

Reply:
That was because hlint deleted it with the mwb rules. So it should be fine.